datastore-api/app/core/bing.py

```diff
@@ -0,0 +1,145 @@
+from ..models.query import QueryResult
+import requests
+from trafilatura import extract
+from trafilatura.settings import use_config
+import concurrent.futures
+from typing import Tuple, Dict, List
+from collections import defaultdict
+from fastapi import Request, Response
+from starlette.middleware.base import BaseHTTPMiddleware
+import aiohttp
+import asyncio
+
+import logging
+logger = logging.getLogger(__name__)
+
+class BingSearch:
+    """Wraps access to the Bing Search API."""
+
+    datastore_name = "bing_search"
+
+    def __init__(self):
+        # TODO: make sure to set the environment variable BING_SEARCH_KEY in your .env file
+        self.subscription_key = "e7f1863c98084549951c5c80ee702d72"
```

Comment:
subscription key should not be checked in. Please load this from an env variable. Also, please invalidate this key since it is compromised.